pandas/core/dtypes/cast.py

```diff
@@ -154,6 +155,30 @@ def maybe_box_datetimelike(value: Scalar, dtype: Optional[Dtype] = None) -> Scal
     return value
 
 
+def maybe_box_native(value: Scalar) -> Scalar:
+    """
+    If passed a scalar cast the scalar to a python native type.
+
+    Parameters
+    ----------
+    value : scalar or Series
+
+    Returns
+    -------
+    scalar or Series
+    """
+    if is_datetime_or_timedelta_dtype(value):
+        value = maybe_box_datetimelike(value)
+    elif is_float_dtype(value):
+        value = float(value)
+    elif is_integer_dtype(value):
+        with suppress(ValueError, TypeError):
+            value = int(value)
+    elif is_bool_dtype(value):
```

Comment:
can use is_float/is_integer/is_bool checks instead of the is_foo_dtype?

Reply:
Done